New input for Crowdstrike Falcon events (#16988)
elastic/beats: x-pack/filebeat/module/crowdstrike/_meta/docs.asciidoc

[role="xpack"]

:modulename: crowdstrike
:has-dashboards: true

== Crowdstrike module

> This is the filebeat module for the Crowdstrike Falcon using the Falcon https://www.crowdstrike.com/blog/tech-center/integrate-with-your-siem[SIEM Connector]. This module collects this data, converts it to ECS, and ingests it to view in the SIEM. By default, the Falcon SIEM connector outputs JSON formated Falcon Streaming API event data in JSON format.

Reviewer: ```suggestion
This is the filebeat module for the Crowdstrike Falcon using the Falcon https://www.crowdstrike.com/blog/tech-center/integrate-with-your-siem[SIEM Connector]. This module collects this data, converts it to ECS, and ingests it to view in the SIEM. By default, the Falcon SIEM connector outputs JSON formatted Falcon Streaming API event data.
```

> This module segments events forwarded by the Falcon SIEM connector into two datasets for endpoint data and Falcon platform audit data.

Reviewer: ```suggestion
```

Author: I was going to keep this since we're still doing this (two datasets), just from the same fileset. That okay?

Reviewer: Oh, that's interesting. I didn't notice that the module was setting the `event.dataset` values. We've never had a case AFAIK where a fileset produced an `event.dataset` value other than the default of `{module}.{fileset}`. I don't think this will cause any issues. I think it would be good to callout the event.dataset values that it produces in this paragraph.

include::../include/what-happens.asciidoc[]

[float]
=== Compatibility

This input supports Crowdstrike Falcon SIEM-Connector-v2.0.

include::../include/running-modules.asciidoc[]

[float]
=== Dashboards

The best way to view Crowdstrike events and alert data is in the SIEM. 

[role="screenshot"]
image::./images/siem-alerts-cs.jpg[]

[float]
For alerts, go to Detections -> External alerts.

[role="screenshot"]
image::./images/siem-events-cs.jpg[]

[float]
And for all over event Crowdstrike Falcon event types, go to Host -> Events.

include::../include/configuring-intro.asciidoc[]

:fileset_ex: falcon_endpoint

include::../include/config-option-intro.asciidoc[]

[float]

> ==== `falcon_endpoint` fileset settings

Reviewer: ```suggestion
==== `falcon` fileset settings
```
